src/ElasticSearchIndexer.php

```diff
@@ -73,43 +77,53 @@ public function getIndexNameByDocument(IndexDocument $document): string
      */
     public function buildIndex(Iterator $documents): int
     {
+        $visited = $skipped = $exceptions = 0;
         $count = 0;
         $blockSize = 0;
         $params = [];
 
-        while ($documents->valid()) {
+        foreach ($documents as $document) {
```

Comment:
Question: Is there a strong reason for iterating this with `while($documents->valid()) + $documents->current()` instead of using a `foreach`? For example, we may not want `rewind()` to be automatically called before starting the iteration, in that case I'll revert the loop back to its former syntax.

Reply:
I think the main reason was to use 'Iterator' features. You just need to be aware about the client of this classes using the Iterator for other purposes, now the pointer will not change anymore after you applied the foreach. 

Btw, I would not refactor unless it is really necessary. I would focus on the log errors. We do not want to increased the test / impact of this code.

Reply:
ok, I changed this but then thought about that possibility. I'll undo this since it is not necessary for our purpose (count the number of iterations). Thanks :)

Reply:
Reverted: dca8412ad80f1bb3a43297f6996151f249dc25a1